ps1/ps1/src/twitter/Extract.java

```diff
@@ -1,177 +0,0 @@
-            char_succeeding = '#';
-
-        } else {
-            // for all other character positions,
-            // we just enter the true preceding/succeeding characters
-            char_preceding = text.charAt(at_char-1);
-            char_succeeding = text.charAt(at_char+1);
-        }
-        
-        // test if it's a username - preceding char is invalid,
-        // and succeeding char is valid
-        
-        if ((AcceptableChars.indexOf(char_preceding) == -1) &&
-            (AcceptableChars.indexOf(char_succeeding) != -1) ){
-                            
-            // get the full string of the username
-            StringBuilder username = new StringBuilder();
-            
-            // add characters to username while characters are valid
-            at_char = at_char + 1;
-            while ((at_char < text.length()) && 
-                    (AcceptableChars.indexOf(text.charAt(at_char)) != -1)){
-                username.append(text.charAt(at_char));
-                at_char = at_char + 1;
-            }
-            
-            return username.toString().toLowerCase();
-            
-        } else {
-            return "";
-        }
-        
-    }
-
-    /**
-     * Get usernames mentioned in a list of tweets.
-     * 
-     * @param tweets
-     *            list of tweets with distinct ids, not modified by this method.
-     * @return the set of usernames who are mentioned in the text of the tweets.
-     *         A username-mention is "@" followed by a Twitter username (as
-     *         defined by Tweet.getAuthor()'s spec). The username-mention cannot
-     *         be immediately preceded or followed by any character valid in a
-     *         Twitter username. For this reason, an email address like
-     *         bitdiddle@mit.edu does NOT contain a mention of the username mit.
-     *         Twitter usernames are case-insensitive, and the returned set may
-     *         include a username at most once.
-     */
-    public static Set<String> getMentionedUsers(List<Tweet> tweets) {
-        
-
-        Set<String> UserList = new HashSet<String>();
-        
-        for (int i = 0; i < tweets.size(); i++){
-                        
-            String t = tweets.get(i).getText();
-            
-            int at_char = 0;
-            at_char = t.indexOf('@', at_char);
-            
-            while (at_char < t.length()) {
-                
-                if(at_char == -1){
-                    at_char = t.length();
-                    continue;
-                } else {
-                    String username = getUsername(t, at_char);
-                    // if not in list, add lowercase username into list
-                    if(!UserList.contains(username) && username != ""){
-                       UserList.add(username);
-                    }
-                    at_char = at_char + username.length() + 1;
-                    at_char = t.indexOf('@', at_char);
```

Comment:
okay i took some time to understand your code, but the two lines above are doing everything. Cool 😄 

That `indexOf` is nice indeed !